pkg/stanza/fileconsumer/file_threadpool.go

```diff
@@ -0,0 +1,198 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package fileconsumer // import "github.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza/fileconsumer"
+
+import (
+	"context"
+	"os"
+	"sync"
+
+	"go.uber.org/zap"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza/fileconsumer/internal/fingerprint"
+)
+
+type readerWrapper struct {
+	reader *reader
+	fp     *fingerprint.Fingerprint
```

Comment:
What's the reason for this struct? Is this fingerprint different than the one inside of the reader?

Reply:
Yeah, it's different. Because the one inside the reader can possibly keep growing as we read to the end.
The one inside struct is the one added to trie, when we first read the file.